src/Uno.UI/UI/Xaml/Controls/Image/Image.wasm.cs

```diff
@@ -88,36 +91,40 @@ private ExceptionRoutedEventArgs ImageFailedConverter(object sender, string e)
 
 			if (e.NewValue is ImageSource source)
 			{
-				_sourceDisposable.Disposable = source.Subscribe(img =>
+				void OnSourceOpened(ImageData img)
 				{
+					_currentImg = img;
 					switch (img.Kind)
 					{
 						case ImageDataKind.Empty:
 							_htmlImage.SetAttribute("src", "");
 							break;
 
-						case ImageDataKind.Base64:
+						case ImageDataKind.DataUri:
 						case ImageDataKind.Url:
 						default:
 							if (MonochromeColor != null)
 							{
-								WebAssemblyRuntime.InvokeJS("Uno.UI.WindowManager.current.setImageAsMonochrome("
-									+ _htmlImage.HtmlId + ", \""
-									+ img.Value + "\", \""
-									+ MonochromeColor.Value.ToHexString() + "\");");
+								WebAssemblyRuntime.InvokeJS("Uno.UI.WindowManager.current.setImageAsMonochrome(" + _htmlImage.HtmlId + ", \"" + img.Value + "\", \"" + MonochromeColor.Value.ToHexString() + "\");");
 							}
 							else
 							{
+								Console.WriteLine($"Setting img src to {img.Value}");
```

Comment:
I agree with the robot

Reply:
![](https://media.giphy.com/media/EX6wKXVM9IOlO/giphy.gif)

Reply:
meaning, delete the `Console.WriteLine`